src/routes/cylinder-set/updateCylinderSet.ts

```diff
@@ -0,0 +1,120 @@
+import { FastifyInstance, FastifyReply, FastifyRequest } from 'fastify';
+import { knexController } from '../../database/database';
+import { errorHandler } from '../../lib/errorHandler';
+import selectCylinderSet from '../../lib/selectCylinderSet';
+import {
+  cylinderSet,
+  CylinderSet,
+  updateCylinderSet,
+  UpdateCylinderSet,
+  UpdateCylinderBody,
+  cylinderSetIdParamsPayload,
+  CylinderSetIdParamsPayload,
+} from '../../types/cylinderSet.types';
+
+const schema = {
+  description: 'Updates a diving cylinder set',
+  tags: ['Cylinder set'],
+  body: updateCylinderSet,
+  params: cylinderSetIdParamsPayload,
+  response: {
+    200: cylinderSet,
+    400: { $ref: 'error' },
+    401: { $ref: 'error' },
+    403: { $ref: 'error' },
+    404: { $ref: 'error' },
+    409: { $ref: 'error' },
+    500: { $ref: 'error' },
+  },
+};
+
+const handler = async (
+  request: FastifyRequest<{
+    Body: UpdateCylinderSet;
+    Params: CylinderSetIdParamsPayload;
+  }>,
+  reply: FastifyReply
+): Promise<void> => {
+  // TODO: Authorization
+
+  const setId = request.params.cylinderSetId;
+
+  // validate cylinder data
+  if (request.body.cylinders !== undefined) {
+    for (const cylinder of request.body.cylinders) {
+      // inspection date
+      if (cylinder.inspection !== undefined) {
+        const inspection = new Date(cylinder.inspection);
+        const thisYear = new Date().getUTCFullYear();
+        if (inspection.getUTCFullYear() > thisYear) {
+          await errorHandler(reply, 400, 'Inspection date from the future');
+          return;
+        }
+      }
+    }
+  }
+
+  await knexController
+    .transaction(async (trx) => {
+      const updateResult = await trx('diving_cylinder_set')
+        .where('id', setId)
+        .update<UpdateCylinderSet>({
+          name: request.body.name,
+          updated_at: trx.fn.now(),
+        });
```

Comment:
Tää päivittää nyt updated_at kenttää, vaikka mitään tietoja ei muutettaisi. Halutaanko tehdä näin?

Reply:
Oikeasti halutaan tehdä siitä kentästä automaattisesti päivittyvä.

Reply:
Tai sitten ei kun tuki sille ei taida olla kovin tietokanta-aknostinen.

Reply:
> Halutaanko tehdä näin?

Kallistun hennosti sille kannalle että haluan.